bounter/count_min_sketch.py

```diff
@@ -96,33 +97,33 @@ def __init__(self, size_mb=64, width=None, depth=None, algorithm='conservative')
             self.width = width
             self.depth = depth
 
-        if algorithm and str(algorithm).lower() == 'log8':
+        if log_counting == 8:
             self.cms = cmsc.CMS_Log8(width=self.width, depth=self.depth)
-        elif algorithm and str(algorithm).lower() == 'log1024':
+        elif log_counting == 1024:
             self.cms = cmsc.CMS_Log1024(width=self.width, depth=self.depth)
-        elif algorithm and str(algorithm).lower() == 'conservative':
+        elif log_counting is None:
             self.cms = cmsc.CMS_Conservative(width=self.width, depth=self.depth)
         else:
-            raise ValueError("Unsupported algorithm! Use 'conservative', 'log8', or 'log1024'!");
+            raise ValueError("Unsupported counting parameter! Use None, 8, or 1024!")
```

Comment:
Let's include the incorrect value in the error message (`%r`), to make it more explicit.